os/drivers/testcase/compression_testcase_drv.c

```diff
@@ -0,0 +1,111 @@
+ * Private Function Prototypes
+ ****************************************************************************/
+
+static int compression_test_drv_ioctl(FAR struct file *filep, int cmd, unsigned long arg);
+static ssize_t compression_test_drv_read(FAR struct file *filep, FAR char *buffer, size_t len);
+static ssize_t compression_test_drv_write(FAR struct file *filep, FAR const char *buffer, size_t len);
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static const struct file_operations compression_test_drv_fops = {
+	0,                            /* open */
+	0,                                                   /* close */
+	compression_test_drv_read,                                /* read */
+	compression_test_drv_write,                               /* write */
+	0,                                                   /* seek */
+	compression_test_drv_ioctl                                /* ioctl */
+#ifndef CONFIG_DISABLE_POLL
+	, 0                                                  /* poll */
+#endif
+};
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/************************************************************************************
+ * Name: compression_test_drv_ioctl
+ *
+ * Description:  The standard ioctl method.
+ *
+ ************************************************************************************/
+
+static int compression_test_drv_ioctl(FAR struct file *filep, int cmd, unsigned long arg)
+{
+	int ret = -EINVAL;
+	/* Handle built-in ioctl commands */
+
+	switch (cmd) {
+	/* ioctl argument:  An integer value indicating the particular test to be run */
+	
+	case TESTIOC_COMPRESSION_TEST:
+		ret = test_compress_decompress(cmd, arg);
+		break;
+	
+	default:
+		lldbg("Unrecognized cmd: %d arg: %ld\n", cmd, arg);
+		break;
+	}
+
+	return ret;
+}
+
+static ssize_t compression_test_drv_read(FAR struct file *filep, FAR char *buffer, size_t len)
+{
+	return 0;                                       /* Return EOF */
+}
+
+static ssize_t compression_test_drv_write(FAR struct file *filep, FAR const char *buffer, size_t len)
+{
+	return len;                                     /* Say that everything was written */
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: compression_test_drv_register
+ *
+ * Description:
+ *   Register /dev/testcase2
+ *
+ ****************************************************************************/
+
+void compression_test_drv_register(void)
+{
+	(void)register_driver(COMPRESSION_TC_DRVPATH, &compression_test_drv_fops, 0666, NULL);
```

Comment:
Could you let me know why you want to make a new driver instead of tc drv?

Reply:
Made the changes